src/main/scala/mesosphere/marathon/core/election/PsuedoElectionStream.scala

```diff
@@ -0,0 +1,18 @@
+package mesosphere.marathon
+package core.election
+
+import akka.actor.Cancellable
+import akka.stream.scaladsl.{ Keep, Source }
+import mesosphere.marathon.stream.EnrichedSource
+
+object PsuedoElectionStream {
+
+  /**
+    * Stream which immediately emits that we are leader; upon cancellation, emits that we are not
+    */
+  def apply(): Source[LeadershipState, Cancellable] = {
+    Source.single(LeadershipState.ElectedAsLeader)
```

Comment:
What do you think about using `Source.maybe(takeWhile)` here?

Reply:
I don't fully understand what you mean. This element needs to yield two elements, one before cancellation, and one after.

Reply:
(I did take your suggestion to make `emptyCancellable` implemented by `Source.maybe`)